src/uct/sm/mm/base/mm_ep.c

```diff
@@ -256,6 +256,13 @@ static UCS_F_ALWAYS_INLINE ssize_t uct_mm_ep_am_common_send(
     UCT_CHECK_AM_ID(am_id);
 
 retry:
+    if (ucs_unlikely(flags & UCT_SEND_FLAG_PEER_CHECK)) {
```

Comment:
maybe do it after posting the message (or detecting NO_RESOURCE) ?

Reply:
will have better latency if doing if after

Reply:
> will have better latency if doing if after

I agree, but can't happen that FIFO is full due to slowness of the receiver and then receiver was killed - so the sender won't be able to detect the issue

Reply:
so we can check in 2 places:
1. after successful send
2. after discovering FIFO is full and cannot send, and going to return NO_RESOURCE

Reply:
good catch, fixed